src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -32,11 +32,11 @@ impl ScalarFunc {
         ctx: &mut EvalContext,
         row: &'a [Datum],
     ) -> Result<Option<Cow<'a, [u8]>>> {
-        let t = try_opt!(self.children[0].eval_time(ctx, row));
+        let t: Cow<'a, Time> = try_opt!(self.children[0].eval_time(ctx, row));
```

Comment:
why we need the type annotation here?

Reply:
Just for supporting auto-completion better in IDE.

Reply:
I think you'd better remove it to make the code style consistent.

Reply:
I think the code style is more consistent by adding type annotation in builtin_time.rs, like `year_week_with_mode`, `year_week_without_mode`, `add_datetime_and_duration`, `add_duration_and_duration` and so on.

Reply:
I know the rust compiler can infer type, but macros like `try_opt!` are not friendly to the IDE's or editor's auto-completion.
If you think these type annotations must be removed, I will. But how about functions like `year_week_with_mode`? Remove type annotation of them together?

Reply:
Thanks for your explanation, it's our fault to make the code style not consistent before. You are right to make it consistent in this file, so we'd like to accept your edition.

Reply:
@brson how do you think about this, is it a good idea to add type annotation here?

Reply:
FYI, in future, we don't need it since null will be treated separately. For now, it's fine.